### PR TITLE
Add test case test_migration_with_unscheduled_replica

### PR DESCRIPTION
**Pull request by chriscchien**
Signed-off-by: Chris Chien <chris.chien@suse.com>

Add test case `test_migration_with_unscheduled_replica`

### CODE REVIEW

File: manager/integration/tests/test_migration.py [hunk 1]
```diff
@@ -123,7 +123,64 @@ def test_migration_with_unscheduled_replica(clients, volume_name):  # NOQA
     11. Enable the scheduling for the node and wait for rebuilding complete.
     12. Validate initially written test data.
     """
-    return
+    # Step 1
+    client = get_random_client(clients)
+
+    # test_nodes[0] : node which running script
+    # test_nodes[1] : migrate target node
+    # test_nodes[2] : node to schedule on/off
```
Comment (khushboo-rancher) [inline manager/integration/tests/test_migration.py:131]: According to your test steps 6-9, the volume is being migrated from node[1] to node[0]. Can we update the description here, target becomes the node[0] and destination becomes node[1].

File: manager/integration/tests/test_migration.py [hunk 2]
```diff
@@ -123,7 +123,64 @@ def test_migration_with_unscheduled_replica(clients, volume_name):  # NOQA
...
+    test_nodes = list()
+    nodes = client.list_node()
+    for node in nodes:
+        if node.name == common.get_self_host_id():
+            test_nodes.append(node)
+    for node in nodes:
+        if node.name != common.get_self_host_id():
+            test_nodes.append(node)
```
Comment (khushboo-rancher) [inline manager/integration/tests/test_migration.py:139]: Are these two iterations on nodes for getting the Longhorn node first, and then other nodes in the list?

**Review by khushboo-rancher — COMMENTED** [review]
LGTM in general, some minor comments.

File: manager/integration/tests/test_migration.py (hunk 2 above)
Comment (chriscchien) [reply manager/integration/tests/test_migration.py:139]: Yes, currently I can not find a more efficient way to do that so I just use 2 loop to come up then I can clearly know every node's role in this test case

File: manager/integration/tests/test_migration.py (hunk 1 above)
Comment (chriscchien) [reply manager/integration/tests/test_migration.py]: Hello @khushboo-rancher, I updated the description, thank you

**Review by khushboo-rancher — APPROVED** [review]

File: manager/integration/tests/test_migration.py (hunk 2 above)
Comment (innobead) [reply manager/integration/tests/test_migration.py:139]: I think you can make the test simpler. WDYT? 

- Use nodes from `get_hosts_for_migration_test` to do the migration testings
- Enable/disable scheduling of any of them
- Just use the local node for the first attachment and write the data

You don't need to prepare a list to store 3 nodes' info, but just reuse an existing function to get the migration nodes to start the testing as above.

**Review by innobead — CHANGES_REQUESTED** [review]

File: manager/integration/tests/test_migration.py (hunk 2 above)
Comment (chriscchien) [reply manager/integration/tests/test_migration.py]: Yes this is more simpler, I updated code by using `get_hosts_for_migration_test` to get nodes  for review

File: manager/integration/tests/test_migration.py [hunk 3]
```diff
@@ -123,7 +123,59 @@ def test_migration_with_unscheduled_replica(clients, volume_name):  # NOQA
...
+
+    # Step 6
+    volume.attach(hostId=hosts[0])
+    volume = common.wait_for_volume_degraded(client, volume_name)
+
+    # Step 7
+    volume.attach(hostId=local_node)
+
+    # Step 8
+    volume = common.wait_for_volume_migration_ready(client, volume_name)
```
Comment (innobead) [inline manager/integration/tests/test_migration.py:162]: You should verify the volume is still degraded and the number of new migrated replicas should the same as the original replicas.

**Review by innobead — CHANGES_REQUESTED** [review]

File: manager/integration/tests/test_migration.py (hunk 3 above)
Comment (chriscchien) [reply manager/integration/tests/test_migration.py:167]: HI @innobead , I update code for checking replica count after migrate, thank you

**Review by innobead — APPROVED** [review]
LGTM

**Comment by innobead** [comment]
@chriscchien Please backport to 1.3 and 1.2. Thanks.

**Pull request merged**